library/spdm_crypt_lib/libspdm_crypt_cert.c

```diff
@@ -1147,11 +1149,13 @@ bool libspdm_x509_common_certificate_check(const uint8_t *cert, size_t cert_size
         goto cleanup;
     }
 
-    /* 10. verify spdm defined extension*/
-    status = libspdm_verify_leaf_cert_spdm_extension(cert, cert_size,
-                                                     is_requester_cert, cert_model);
-    if (!status) {
-        goto cleanup;
+    if ((!set_cert) || (cert_model == SPDM_CERTIFICATE_INFO_CERT_MODEL_DEVICE_CERT)) {
```

Comment:
If it's
- `SET_CERTIFICATE` AND
- Responder supports `ALIAS_CERT` AND
- Device Certificate CA contains a hardware identity

then libspdm should check that it matches the SPDM hardware identity OID in `libspdm_verify_leaf_cert_spdm_extension`.

Reply:
> * `SET_CERTIFICATE` AND
> 
> * Responder supports `ALIAS_CERT` AND
> 
> * Device Certificate CA contains a hardware identity

This is invalid, "If the AliasCert model is used, this OID shall not be present in any alias certificates in the chain".

Are you saying we should check that the hardware identity OID isn't included?

Reply:
![image](https://github.com/DMTF/libspdm/assets/8474235/69a74f04-afdf-48e2-a353-ac858b4c6589)
Alias certificates start right after the Device Certificate CA. In particular
>When the AliasCert model is in use, the Device Certificate is referred to as a Device Certificate CA, indicating that the certificate both contains device hardware identity information and functions as a certificate authority to sign an additional certificate. In this model, the Device Certificate CA should contain the [Hardware identity OID](https://github.com/DMTF/SPDM-WG/blob/master/docs/DSP0274/SPDMSpecification.md#hardware-identity-oid).

But reading that again it's a `should` not a `shall`. However `libspdm_verify_leaf_cert_spdm_extension` can be used in the future for other checks besides the hardware identity. Maybe add the `set_cert` parameter to `libspdm_verify_leaf_cert_spdm_extension` and move this logic to the function. That way `libspdm_verify_leaf_cert_spdm_extension` can be called unconditionally and all the set_cert / cert_model logic can be handled in the function.

Reply:
I dont think we need such change at this moment.

`libspdm_verify_leaf_cert_spdm_extension` is to verify "leaf_cert", but in set_cert case, the last one is not leaf_cert.

I think current code is good enough - just skip the leaf_cert verification.

Reply:
> but in set_cert case, the last one is not leaf_cert.

libspdm does check the Device Certificate CA's CA field via https://github.com/DMTF/libspdm/blob/0ef69e7919f6949b39355849b2116b33d8167334/library/spdm_crypt_lib/libspdm_crypt_cert.c#L1326-L1328 The documentation for that function is not correct. I'll correct it in a pull request.

>I think current code is good enough - just skip the leaf_cert verification.

Ok but long-term this should be refactored as the logic is starting to get confusing.